tests/integration/test_integration.py

```diff
@@ -219,6 +219,9 @@ def test_metrics(encoding, monkeypatch):
 
     with override_global_config(dict(health_metrics_enabled=True)):
         t = Tracer()
+        t.configure(
```

Comment:
We will need to duplicate this test so we have validation of the default behavior and when partial flushing is disabled.

That is, duplicate `test_metrics` so we have `test_metrics_partial_flush_disabled`.

`test_metrics` should not call `t.configure(partial_flush_enabled=False)` and then we need to update the assertions to match what we expect the default behavior to be.

Then `test_metrics_partial_flush_disabled` can call `t.configure(partial_flush_enabled=False)` and have the same assertions that exist today.